feat(flutter): Add auth next steps (#5675)
aws-amplify/docs: src/fragments/lib/auth/flutter/signin_next_steps/30_confirm_custom_challenge.mdx

If the next step is `confirmSignInWithCustomChallenge`, Amplify Auth is awaiting completion of a custom authentication challenge. 
The challenge is based on the Lambda trigger you configured as part of a [custom sign in flow](/lib/auth/signin_with_custom_flow).

For example, your custom challenge Lambda may pass a prompt to the frontend which requires the user to enter a secret code.

```dart
Future<void> _handleSignInResult(SignInResult result) async {
  switch (result.nextStep.signInStep) {
    // ...
    case AuthSignInStep.confirmSignInWithCustomChallenge:
      final parameters = result.nextStep.additionalInfo;
      final hint = parameters['hint']!;
      safePrint(hint); // "Enter the secret code"
    // ...
  }
}
```

> To complete this step, you should prompt the user for the custom challenge answer, and pass the answer to the `confirmSignIn` API.

Reviewer: ```suggestion
To complete this step, prompt the user for the custom challenge answer, and pass the answer to the `confirmSignIn` API.
```

```dart
Future<void> confirmCustomChallenge(String answer) async {
  try {
    final result = await Amplify.Auth.confirmSignIn(
      confirmationValue: answer,
    );
    return _handleSignInResult(result);
  } on AuthException catch (e) {
    safePrint('Error confirming custom challenge: ${e.message}');
  }
}
```

<Callout warning>

<b>Special Handling on confirmSignIn</b>

> During a confirmSignIn call if `failAuthentication=true` is returned by the Lambda, the session of the request gets invalidated by Cognito.
> In this case, a `NotAuthorizedException` is thrown and you are required to restart the sign-in flow by calling `Amplify.Auth.signIn`.

Reviewer: ```suggestion
If `failAuthentication=true` is returned by the Lambda, Cognito will invalidate the session of the request.
This is represented by a `NotAuthorizedException`, which requires restarting the sign-in flow by calling `Amplify.Auth.signIn`.
```
